compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -644,6 +646,15 @@ fn print_crate_info(
             TargetSpec => {
                 println!("{}", serde_json::to_string_pretty(&sess.target.to_json()).unwrap());
             }
+            AllTargetSpecs => {
+                let mut targets = BTreeMap::new();
+                for name in rustc_target::spec::TARGETS.to_vec() {
+                    let triple = TargetTriple::from_triple(name);
+                    let Ok((target, _)) = Target::search(&triple, &sess.sysroot) else { continue };
```

Comment:
Is there a particular reason this uses `search` instead of loading it directly, like:

```suggestion
                    let target = Target::expect_builtin(&triple);
```

Reply:
Not really, not much familiar with this part of the codebase and that was the method I found doing a brief search.